### PR TITLE
add CITATION.cff 

### PR DESCRIPTION
**Pull request by martenson**
This PR proposes including a new github-supported CITATION format in the root directory. Before merging we should discus:

* whether we want to support this format
* what should the contents look like (this PR is trying to mirror https://github.com/galaxyproject/galaxy/blob/dev/CITATION)
* whether we replace CITATION file with it

Github integration described at https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/creating-a-repository-on-github/about-citation-files
Github integration preview at https://github.com/martenson/galaxy
Standard at https://citation-file-format.github.io/ and https://citation-file-format.github.io/assets/pdf/cff-specifications-1.0.3.pdf

fyi github generates these citation strings

bibtex
```
@misc{Afgan_The_Galaxy_platform_2018,
author = {Afgan, Enis and Baker, Dannon and Batut, Bérénice and van den Beek, Marius and Bouvier, Dave and Čech, Martin and Chilton, John and Clements, Dave and Coraor, Nate and Grüning, Björn and Guerler, Aysam and Hillman-Jackson, Jennifer and Hiltemann, Saskia and Jalili, Vahid and Rasche, Helena and Soranzo, Nicola and Goecks, Jeremy and Taylor, James and Nekrutenko, Anton and Blankenberg, Daniel},
doi = {10.1093/nar/gky379},
month = {5},
title = {The Galaxy platform for accessible, reproducible and collaborative biomedical analyses: 2018 update},
url = {https://doi.org/10.1093/nar/gky379},
year = {2018}
}
```

apa
```
Afgan E., Baker D., Batut B., van den Beek M., Bouvier D., Čech M., Chilton J., Clements D., Coraor N., Grüning B., Guerler A., Hillman-Jackson J., Hiltemann S., Jalili V., Rasche H., Soranzo N., Goecks J., Taylor J., Nekrutenko A., Blankenberg D. (2018). The Galaxy platform for accessible, reproducible and collaborative biomedical analyses: 2018 update (version 18.05). DOI: https://doi.org/10.1093/nar/gky379
```

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  - family-names: "Goecks"
+    given-names: "Jeremy"
+  - family-names: "Taylor"
+    given-names: "James"
+  - family-names: "Nekrutenko"
+    given-names: "Anton"
+  - family-names: "Blankenberg"
+    given-names: "Daniel"
+title: "The Galaxy platform for accessible, reproducible and collaborative biomedical analyses: 2018 update"
+version: 18.05
```
Comment (martenson) [inline CITATION.cff:45]: This is a required attribute, unclear what should be there for us. For now I've added the release at the time of publication.

File: CITATION.cff [hunk 2]
```diff
@@ -0,0 +1,48 @@
+cff-version: 1.2.0
+message: "If you use or extend Galaxy in your published work, please cite this publication as below"
+authors:
```
Comment (nsoranzo) [inline CITATION.cff:3]: This should be the list of authors of the software, not of the paper, here we could use an [entity](https://github.com/citation-file-format/citation-file-format/blob/main/README.md#entity-objects) with `name: Galaxy committers`.
The authors, title, journal, volume, number, start, end of the paper should go under the [references](https://github.com/citation-file-format/citation-file-format/blob/main/README.md#references-optional) key.

File: CITATION.cff [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    given-names: "Jeremy"
+  - family-names: "Taylor"
+    given-names: "James"
+  - family-names: "Nekrutenko"
+    given-names: "Anton"
+  - family-names: "Blankenberg"
+    given-names: "Daniel"
+title: "The Galaxy platform for accessible, reproducible and collaborative biomedical analyses: 2018 update"
+version: 18.05
+doi: 10.1093/nar/gky379
```
Comment (nsoranzo) [inline CITATION.cff:46]: This should also be the doi of the software if deposited on e.g. Zenodo.

File: CITATION.cff [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+date-released: 2018-05-22
+url: "https://doi.org/10.1093/nar/gky379"
```
Comment (nsoranzo) [inline CITATION.cff:48]:
```suggestion
url: "https://github.com/galaxyproject/galaxy"
```

File: CITATION.cff [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    given-names: "Nicola"
+  - family-names: "Goecks"
+    given-names: "Jeremy"
+  - family-names: "Taylor"
+    given-names: "James"
+  - family-names: "Nekrutenko"
+    given-names: "Anton"
+  - family-names: "Blankenberg"
+    given-names: "Daniel"
+title: "The Galaxy platform for accessible, reproducible and collaborative biomedical analyses: 2018 update"
```
Comment (nsoranzo) [inline CITATION.cff:44]:
```suggestion
title: "Galaxy"
```

File: CITATION.cff (hunk 2 above)
Comment (martenson) [reply CITATION.cff:3]: This project has many more authors than `committers` group (see https://github.com/galaxyproject/galaxy/blob/dev/CONTRIBUTORS.md) -- at that point the citation will become unwieldy I think.

Additionally afaik we want to ask people to cite the papers, if the github widget offers disconnected repository citation instead I'd rather not have it I think.

**Comment by martenson** [comment]
xref https://github.com/galaxyproject/galaxy/pull/15123

**Pull request closed without merging**